File: 414-third-maximum-number/third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        
        List<Integer> al = new ArrayList<>();
        for (int i=0; i<nums.length; i++) {
            if (!al.contains(nums[i]))
                al.add(nums[i]);
        }
        if (al.size() <= 2){
            return Collections.max(al);
        }

        int least = Integer.MIN_VALUE;
        int first = least, second = least, third = least;
        for (int i : al) {
            if (i > first){
                third = second;
                second = first;
                first = i;
            } else if (i > second) {
                third = second;
                second = i;
            } else if (i > third) {
                third = i;
            }
        }
        return third;
    }
}

File: 414-third-maximum-number/third-maximum-number.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        nums.erase(unique(nums.begin(), nums.end()), nums.end());
        int n = nums.size();

        if (n < 3) {
            return nums[n - 1];
        } else {
            return nums[n - 3];
        }
    }
};